extension/solarwindsapmsettingsextension/extension.go

```diff
@@ -75,7 +90,91 @@ func (extension *solarwindsapmSettingsExtension) Shutdown(_ context.Context) err
 	return nil
 }
 
-func refresh(extension *solarwindsapmSettingsExtension) {
-	// Concrete implementation will be available in later PR
-	extension.logger.Info("refresh task")
+func refresh(extension *solarwindsapmSettingsExtension, filename string) {
+	extension.telemetrySettings.Logger.Info("time to refresh", zap.String("endpoint", extension.config.ClientConfig.Endpoint))
+	if hostname, err := os.Hostname(); err != nil {
+		extension.telemetrySettings.Logger.Error("unable to call os.Hostname()", zap.Error(err))
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), grpcContextDeadline)
```

Comment:
I would suggest passing context in or storing some async state that you can tell if you're meant to shutdown so it doesn't either leak routines or blocks un-neededly.

It is a lot easier to pass in the context here

Reply:
The context is for the gRPC request to Solarwinds APM backend. 

We don't want any context timeout / cancel of the request causing any cancellation / error to collector as we can always refresh in the next cycle.